ambry-router/src/main/java/com.github.ambry.router/SimpleOperationTracker.java

```diff
@@ -66,27 +69,32 @@
 class SimpleOperationTracker implements OperationTracker {
   protected final String datacenterName;
   protected final String originatingDcName;
-  protected final int successTarget;
-  protected final int parallelism;
+  protected final int diskSuccessTarget;
+  protected final int diskParallelism;
+  protected final int cloudSuccessTarget;
+  protected final int cloudParallelism;
+  protected final boolean cloudReplicasPresent;
+  // How many NotFound responses from originating dc will terminate the operation.
+  // It's decided by the success target of each mutation operations, including put, delete, update ttl etc.
+  protected final int originatingDcNotFoundFailureThreshold;
+  protected final int totalReplicaCount;
   protected final LinkedList<ReplicaId> replicaPool = new LinkedList<>();
 
-  protected int totalReplicaCount = 0;
   protected int inflightCount = 0;
-  protected int succeededCount = 0;
+  protected int diskSucceededCount = 0;
+  protected int cloudSucceededCount = 0;
```

Comment:
I would suggest renaming these variables to something like  `diskReplicaSuccessCount` to distinguish it from `diskDownCount` more easily.